ocs_ci/ocs/platform_nodes.py

```diff
@@ -818,6 +824,271 @@ def _prepare_node(self, node_id):
+        pk = 'ParameterKey'
+        pv = 'ParameterValue'
+
+        param_list.append({pk: 'Index', pv: index})
+        param_list.append({pk: 'InfrastructureName', pv: self.infra_id})
+        param_list.append({pk: 'RhcosAmi', pv: self.worker_image_id})
+        param_list.append(
+            {
+                pk: 'IgnitionLocation', pv: self.worker_ignition_location
+            }
+        )
+        param_list.append({pk: 'Subnet', pv: self.worker_subnet})
+        param_list.append(
+            {
+                pk: 'WorkerSecurityGroupId', pv: self.worker_security_group
+            }
+        )
+        param_list.append(
+            {
+                pk: 'WorkerInstanceProfileName', pv: self.worker_instance_profile
+            }
+        )
+        param_list.append(
+            {
+                pk: 'WorkerInstanceType', pv: conf['worker_instance_type']
+            }
+        )
+
+        return param_list
+
+    def add_cert_to_template(self, worker_template_path):
+        """
+        Add cert to worker template
+
+        Args:
+            worker_template_path (str): Path where template file is located
+
+        """
+        worker_ignition_path = os.path.join(
+            self.cluster_path,
+            constants.WORKER_IGN
+        )
+        cert = self.get_cert_content(worker_ignition_path)
+        self.update_template_with_cert(worker_template_path, cert)
+
+    @retry(exceptions.PendingCSRException, tries=4, delay=10, backoff=1)
+    def approve_pending_csr(self):
+        """
+        After node addition csr could be in pending state,we have to
+        approve it
+
+        Raises:
+            exceptions.PendingCSRException
+
+        """
+        cmd = "adm certificate approve"
+        self.get_csr_resource()
+        for item in self.csr_conf.data.get('items'):
+            cmd = f"{cmd} {item.get('metadata').get('name')}"
+            self.csr_conf.ocp.exec_oc_cmd(cmd)
+
+        try:
+            self.check_no_pending_csr()
+            logger.info("All the csrs approved")
+        except exceptions.PendingCSRException:
+            logger.error("Failed to approve all the CSR")
+            raise
+
+    @retry(exceptions.PendingCSRException, tries=2, delay=300, backoff=1)
+    def check_no_pending_csr(self):
+        """
+        Check whether we have any pending csrs
+
+        Raises:
+            exceptions.PendingCSRException
+
+        """
+        # Load the latest state of csr
+        self.csr_conf.get()
+        pending = False
+        for item in self.csr_conf.data.get('items'):
+            if item.get('status') == {}:
+                logger.warning(f"{item.get('metadata').get('name')} is not Approved")
+                pending = True
+        if pending:
+            raise exceptions.PendingCSRException("Some of the csrs are in 'Pending' state")
+
+    def get_csr_resource(self):
+        """
+        Get the csr data into OCS object
+        """
+        self.csr_conf = ocs.OCS(
```

Comment:
This functionality will be moved out to the utils as it will have to be used for deployment as well. I suggest to make it simpler and just return value instead of use assign this to self.csr_conf

I think that you are anyway trying to always reloading the data. If you will just have functionality which will return just items will be maybe enough for us WDYT?  Also do you think that we have to wrap it in OCS object? Isn't  OCP object enough for this purpose?

Reply:
ya, I didn't see reload() function in OCP and wanted to use reload from OCS but ultimately I ended up using plain get(). So yep we can use OCP itself.